### PR TITLE
needs review

### PR DESCRIPTION
**Pull request by ghost**
There are a lot of changes.

* fix memory bloat
* background preload
* can save circular reference
* fix box calculation
* CSS Font loading

any questions are welcome!!

### CODE REVIEW

**Comment by hajimehoshi** [comment]
I'd split this into some PRs to make it easier to review.

**Comment by krmbn0576** [comment]
I have not seen it in detail yet, but I have some questions.

- Why is "CSS Font Loading" necessary? Is not the current font loading insufficient?
- "Fix memory bloat" and "background preload" are very huge changes, so I would like a brief explanation.

**Comment by ghost** [comment]
@hajimehoshi 
Hmm, how? I have no idea to do this.

@krmbn0576 
> Why is "CSS Font Loading" necessary? Is not the current font loading insufficient?

Old algorithm cannot handle errors, and force failing if loading takes over 20 sec.
This may occur under poor connection.

> "Fix memory bloat" and "background preload" are very huge changes, so I would like a brief explanation.

I'll write it later (heavy work for me ; v ;)

**Comment by hajimehoshi** [comment]
> Hmm, how? I have no idea to do this.

Just make multiple PRs for individual changes. Is this hard?

**Comment by ghost** [comment]
Discard this branch? I see. I'll try.

**Comment by hajimehoshi** [comment]
Yeah, IMHO this PR is too big to review...

File: plugin/Atsumaru_Debug_ReportMemory.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    function updateInfo(){
+        var content = 'size: ' + Math.floor(ImageManager._imageCache.getSize()/1000)/1000 + 'MPix<br>';
+        content += 'totalCount: ' + ImageManager._imageCache.countBitmap() + '<br>';
+        content += 'requestCount: ' + ImageManager._imageCache.countRequest() + '<br>';
+
+        div.innerHTML = content;
+        div.style.zIndex = 11;
+    }
+
+    const ImageManager_update = ImageManager.update;
```
Comment (hiroppy) [inline plugin/Atsumaru_Debug_ReportMemory.js:18]: I think you should use `var`.

Comment (ghost) [reply plugin/Atsumaru_Debug_ReportMemory.js:18]: ooh mistake
thanks

**Pull request closed without merging**